src/main/java/com/uber/jenkins/phabricator/coverage/CoverageProvider.java

```diff
@@ -85,12 +56,33 @@ String getCoverageReportPattern() {
      * Get the coverage metrics for the provider
      * @return The metrics, if any are available
      */
+    @Nullable
     public CodeCoverageMetrics getMetrics() {
         if (!hasCoverage()) {
             return null;
         }
         return getCoverageMetrics();
     }
 
+    @Nullable
+    String getRelativePathFromProjectRoot(String file) {
+        return getRelativePathFromProjectRoot(includeFiles, file);
+    }
+
+    @Nullable
+    static String getRelativePathFromProjectRoot(Map<String, String> includeFiles, String file) {
+        String finalFile = null;
+        if (includeFiles == null || includeFiles.isEmpty()) {
+            finalFile = file;
+        } else {
+            String includedFileName = FilenameUtils.getName(file);
```

Comment:
That is the same case even before with cobertura and they way it used to guess the filename based on the package name + class name (what jacoco also does today). The issue is with languages like kotlin/Scala which can have multiple top level classes in a file, they will not be able to get code coverage since package path + source file name will not match where they are actually present in the filesystem. 

A better approach might be to do two separate marches i.e one with just file name and another with package name + sourcefile name. This way multiple classes with same name and different packages will match correctly. In the case the full match is not possible, we fall back to South file name match and pick just one. The fact that we only check against files part of a diff means that the possibility of a bad match is very unlikely (only if two files with same name are touched as part of the diff), but that will the beat we can do.